sdk/cognitivelanguage/ai-language-text/test/public/utils/customTestHelpter.ts

```diff
@@ -0,0 +1,171 @@
+    }
+  }
+  rm(dirName, { recursive: true, force: true });
+}
+
+async function getAssetsforProject(
+  projectKind:
+    | "CustomEntityRecognition"
+    | "CustomSingleLabelClassification"
+    | "CustomMultiLabelClassification"
+): Promise<{
+  assets:
+    | ExportedCustomSingleLabelClassificationProjectAssets
+    | ExportedCustomMultiLabelClassificationProjectAssets
+    | ExportedCustomEntityRecognitionProjectAssets;
+  dirName: string;
+}> {
+  switch (projectKind) {
+    case "CustomSingleLabelClassification":
+      return { assets: customSingleLabelAssets, dirName: path.join(pathName, "WebOfScience") };
+
+    case "CustomMultiLabelClassification":
+      return { assets: customMultiLabelAssets, dirName: path.join(pathName, "MoviesSummary") };
+
+    case "CustomEntityRecognition":
+      return { assets: customEntityAssets, dirName: path.join(pathName, "LoanAgreements") };
+  }
+}
+
+async function polling(client: TextAuthoringClient, project: any) {
+  const poller = getLongRunningPoller(client, project);
+  await poller.pollUntilDone();
+}
+
+export async function createCustomTestProject(
+  client: TextAuthoringClient,
+  projectKind:
+    | "CustomEntityRecognition"
+    | "CustomSingleLabelClassification"
+    | "CustomMultiLabelClassification",
+  projectName: string,
+  deploymentName: string
+): Promise<void> {
+  const { assets, dirName } = await getAssetsforProject(projectKind);
+  await decompress(dirName.concat(".zip"), pathName);
+  await uploadDocumentsToStorage(dirName);
+
+  const createProjectOptions: CreateProjectOptions = {
+    projectKind,
+    storageInputContainerName,
+    projectName,
+    language,
+    multilingual: false,
+    settings: {},
+  };
+
+  // Create a project
+  const importTask = await client
+    .path("/authoring/analyze-text/projects/{projectName}/:import", projectName)
+    .post({
+      body: {
+        metadata: createProjectOptions,
+        projectFileVersion: "2022-05-01",
+        stringIndexType: "Utf16CodeUnit",
+        assets,
+      },
+    });
+  await polling(client, importTask);
+  logger.info("Project created", projectName);
+
+  // Start training
+  const trainTask = await client
+    .path("/authoring/analyze-text/projects/{projectName}/:train", projectName)
+    .post(trainParam);
+  await polling(client, trainTask);
+  logger.info("Training finished");
+
+  // Deploy model
+  const deployTask = await client
+    .path(
+      "/authoring/analyze-text/projects/{projectName}/deployments/{deploymentName}",
+      projectName,
+      deploymentName
+    )
+    .put(deployParam);
+  await polling(client, deployTask);
+  logger.info("Model deployment finished");
+}
+
+export async function cleanupCustomTestResource(
+  client: TextAuthoringClient,
+  projectName: string
+): Promise<void> {
+  const deleteTask = await client
+    .path("/authoring/analyze-text/projects/{projectName}", projectName)
+    .delete();
+  await polling(client, deleteTask);
```

Comment:
do we need to wait for the deletion to complete?

Reply:
We don't, I'll update this